### PR TITLE
Support generating converters for all objects

### LINKED ISSUE #1235: Create converters for all types, not just top level
Unless `--just-types` is specified, a handy `Convert` utility is created to cast raw JSON to well typed data models. However, these are only created for the top level types.

Support creating conversion routines for all types described in a schema.

This will support the common scenario where REST APIs return nested resources. As a fabricated example:
* The JSON schema `car.json` describes the top level `Car` type and a nested `CarManufacturer` type.
* The API provides a top level `car/:carId` resource. This maps to the quicktype `Convert.toCar` helper ✅ 
* The API provides a nested `car/:carId/manufacturer` resource. While quicktype created the type for us, there is no matching converter ❌

### PR DESCRIPTION
**Pull request by benquarmby**
Resolves #1235

### CODE REVIEW

File: src/quicktype-core/language/JavaScript.ts [hunk 1]
```diff
@@ -273,7 +276,17 @@ export class JavaScriptRenderer extends ConvenienceRenderer {
                     this.emitLine("return JSON.stringify(uncast(value, ", typeMap, "), null, 2);");
                 }
             });
-        });
+        };
+
+        switch (this._jsOptions.converters) {
+            case ConvertersOptions.All:
+                this.forEachObject("interposing", converter);
```
Comment (benquarmby) [inline src/quicktype-core/language/JavaScript.ts]: #naming-is-hard

Is `All` the right name for this? My assumption is this wont include enums?

File: src/quicktype-core/support/Converters.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { EnumOption } from "../RendererOptions";
+
+export enum ConvertersOptions {
+    Top = "top",
```
Comment (schani) [inline src/quicktype-core/support/Converters.ts]: I'd call this `TopLevel = "top-level"`.

**Comment by schani** [comment]
Yes, this can ship with just JS/TS/Flow.

I left a comment on naming.

Right now the only easy test you can have is by adding the new option to the `quickTestRendererOptions`.  When the quicktype tests run, all input files are tested with default options, but then a small subset of input files is also tested with all of the `quickTestRendererOptions`.  In this case none of the additional generated code would run, but it would at least have to compile.  To test more than that you'd have to write custom tests, for which we don't have a nice framework yet.

**Comment by schani** [comment]
There are [some differences in the output](https://s3.amazonaws.com/quicktype-outputs/diffs/53fc1c6bbba6ce2b38bd404278e2fae8b53901c8-7e40e8fc2ffb6afe8884a6bb4e512284205005ce.diff) of GraphQL tests.  Any idea why they happen?

**Comment by benquarmby** [comment]
Will rename the `TopLevel` option and try out the `quickTestRendererOptions`.

Can't explain that diff. Will comb over the code changes again.

File: src/quicktype-core/language/JavaScript.ts (hunk 1 above)
Comment (benquarmby) [reply src/quicktype-core/language/JavaScript.ts]: Renamed to `AllObjects`.

File: src/quicktype-core/support/Converters.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import { EnumOption } from "../RendererOptions";
+
+export enum ConvertersOptions {
+    TopLevel = "top-level",
```
Comment (benquarmby) [inline src/quicktype-core/support/Converters.ts]: Renamed to `TopLevel`.

**Comment by schani** [comment]
Fantastic, thank you!

**Pull request merged**